File: session6/lecture151/beakjoon_1202_me.cpp
#include <bits/stdc++.h>
using namespace std;

int n, k, c, ret;

vector<pair<int, int>> temps;
vector<pair<int, int>> bags;

int main()
{
  cin.tie(NULL);

  cin >> n >> k;

  for (int i = 0; i < n; i++)
  {
    int m, v;
    cin >> m >> v;
    temps.push_back({m, v});
  }

  cin >> c;

  for (pair<int, int> t : temps)
  {
    if (t.first <= c)
    {
      bags.push_back(t);
    }
  }

  sort(
      bags.begin(), bags.end(),
      [](const auto &a, const auto &b)
      {
        return a.second < b.second;
      });

  for (int i = 0; i < c; i++)
  {
    ret += bags[i].second;
  }
  cout << ret;
}

File: session6/lecture151/beakjoon_1202_me.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8').trim().split('\n');
let line = 0;
const readLine = () => input[line++];

let ret = 0;

const [n, k] = readLine().split(' ').map(Number);

const temps = [];
const bags = [];

for (let i = 0; i < n; i++) {
  const [m, v] = readLine().split(' ').map(Number);
  temps.push([m, v]);
}

const c = Number(readLine());

for (const t of temps) {
  if (t[0] <= c) {
    bags.push(t);
  }
}

bags.sort((a, b) => a[1] - b[1]);

for (let i = 0; i < c; i++) {
  ret += bags[i][1];
}
console.log(ret);
